### ISSUE TITLE
Pagination-generated pages driven by custom collections are lazily added to built-in collections

### ISSUE THREAD

**Issue by hispanic** [11ty/eleventy#1321]
The built-in collections (i.e., collections.all and all tag-specific collections) grow larger with the generation of each additional pagination-driven page.

Create three pagination-driven pages (e.g., p8n1.njk, p8n2.njk, and p8n3.njk), each driven by a custom collection (e.g., collections.p8n1Coll, collections.p8n2Coll, and collections.p8n3Coll). Each custom collection is programmatically registered using the eleventyConfig.addCollection() function and created by filtering the results of the collectionApi.getAll() function. 

At build-time (i.e., npx @11ty/eleventy), observe the following:
1. During the process of creating the first of these custom collections (e.g., collections.**p8n1**Coll), the array returned by collectionApi.getAll() does **not** contain any of the three pagination-driven pages.
2. During the process of creating the second of these custom collections (e.g., collections.**p8n2**Coll), the array returned by collectionApi.getAll() now **does** contain the **p8n1** pagination-driven page, but does **not** contain the second or third pagination-driven pages.
3. During the process of creating the third of these custom collections (e.g., collections.**p8n3**Coll), the array returned by collectionApi.getAll() **now** contains the **p8n1** and **p8n2** pagination-driven pages, but does **not** contain the third pagination-driven page.
4. A subsequent call to collectionApi.getAll() **now** contains **all three** pagination-driven pages.

In short, the built-in collections (i.e., collections.all and, if applicable, all tag-specific collections) grow larger with the generation of each additional pagination-driven page. 

I can understand why this is, but it can be a bit frustrating to deal with. I stumbled on this as I was trying to build a sitemap page. I noticed that the results weren't including my recent-posts page. (The custom collection driving the creation of the sitemap pagination is getting created **before** the custom collection driving the creation of the recent-posts page.) 

It would help if there was some way to dictate the order of creation for each custom collection. Of course, a more elegant solution would be welcome, too.

I'm using Eleventy v0.11.0 on MacOS.

Thank you for Eleventy! :)

**Comment by zachleat** [comment]
We’re a few years in now and I’d guess that this really old bug report is _likely_ fixed

But unfortunately this issue slipped under the radar—I’m finally circling back through some of the really old issues on the tracker and am unfortunately declaring issue bankruptcy on these—sorry! If you feel like this issue is still relevant, feel free to reopen a new issue with the same exact content! Appreciate your understanding!
